core/chain_manager.go

```diff
@@ -92,6 +92,7 @@ type ChainManager struct {
 	futureBlocks *BlockCache
 
 	quit chan struct{}
+	wg   sync.WaitGroup
```

Comment:
First of all, these sort of changes need to be well documented. Second I suggest you find some other way to solve this.